pkg/network/encoding/encoding.go

```diff
@@ -58,30 +53,51 @@ func GetUnmarshaler(ctype string) Unmarshaler {
 	return jSerializer
 }
 
-func modelConnections(conns *network.Connections) *model.Connections {
-	cfgOnce.Do(func() {
-		agentCfg = &model.AgentConfiguration{
+// ConnectionsModeler contains all the necessary structs for modeling a connection.
+type ConnectionsModeler struct {
+	httpEncoder                 *httpEncoder
+	http2Encoder                *http2Encoder
+	kafkaEncoder                *kafkaEncoder
+	connTelemetryMap            map[string]int64
+	compilationTelemetryByAsset map[string]*model.RuntimeCompilationTelemetry
+	kernelHeaderFetchResult     model.KernelHeaderFetchResult
+	coreTelemetryByAsset        map[string]model.COREResult
+	agentCfg                    *model.AgentConfiguration
+	prebuiltEBPFAssets          []string
+	batchIndex                  int
+}
+
+// InitConnectionsModeler initialize the connection modeler with the encoders, telemetry, and agent configuration of
+// the current connections.
+func InitConnectionsModeler(conns *network.Connections) *ConnectionsModeler {
```

Comment:
idiomatic Go generally has constructors named `New...` rather than `Init...`.

Also some documentation on what is expected to be in the arguments of the constructor versus `ModelConnections`, since it isn't immediately clear how to use this object.

Reply:
I have rename the function, as well I have tried to improve the current doc for the function and the use of the arguments inside the object